File: backend/src/main/java/com/example/cinequest/entity/Role.java
package com.example.cinequest.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity(name = "role")
public class Role {
    @Id
    @Column(name = "name")
    String name;

    @Column(name = "description")
    String description;
}

File: frontend/lib/src/domain/auth/entities/role.dart
import 'package:equatable/equatable.dart';

class Role extends Equatable {
  const Role({
    required this.name,
    this.description,
  });

  final String name;
  final String? description;

  @override
  List<Object?> get props => [
        name,
        description,
      ];
}
